File: frontend/lib/core/models/order.dart
import 'branch.dart';
import 'order_item.dart';

class Order {
  final int id;
  final String orderNumber;
  final String orderStatus;
  final String paymentStatus;
  final String paymentMethod;
  final String? notes;
  final int subtotal;
  final int serviceFee;
  final int total;
  final String? paidAt;
  final String? midtransSnapToken;
  final String createdAt;
  final String updatedAt;
  final Branch? branch;
  final List<OrderItem>? items;

  Order({
    required this.id,
    required this.orderNumber,
    required this.orderStatus,
    required this.paymentStatus,
    required this.paymentMethod,
    this.notes,
    required this.subtotal,
    required this.serviceFee,
    required this.total,
    this.paidAt,
    this.midtransSnapToken,
    required this.createdAt,
    required this.updatedAt,
    this.branch,
    this.items,
  });

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      id: json['id'] as int,
      orderNumber: json['order_number'] as String,
      orderStatus: json['order_status'] as String,
      paymentStatus: json['payment_status'] as String,
      paymentMethod: json['payment_method'] as String,
      notes: json['notes'] as String?,
      subtotal: json['subtotal'] as int,
      serviceFee: json['service_fee'] as int? ?? 2000,
      total: json['total'] as int,
      paidAt: json['paid_at'] as String?,
      midtransSnapToken: json['midtrans_snap_token'] as String?,
      createdAt: json['created_at'] as String,
      updatedAt: json['updated_at'] as String,
      branch: json['branch'] != null ? Branch.fromJson(json['branch']) : null,
      items: json['items'] != null
          ? (json['items'] as List)
              .map((i) => OrderItem.fromJson(i as Map<String, dynamic>))
              .toList()
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'order_number': orderNumber,
      'order_status': orderStatus,
      'payment_status': paymentStatus,
      'payment_method': paymentMethod,
      'notes': notes,
      'subtotal': subtotal,
      'service_fee': serviceFee,
      'total': total,
      'paid_at': paidAt,
      'midtrans_snap_token': midtransSnapToken,
      'created_at': createdAt,
      'updated_at': updatedAt,
      'branch': branch?.toJson(),
      'items': items?.map((i) => i.toJson()).toList(),
    };
  }
}

File: backend/app/Models/Order.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Order extends Model
{
    protected $fillable = [
        'user_id',
        'branch_id',
        'order_number',
        'order_status',
        'payment_status',
        'payment_method',
        'notes',
        'subtotal',
        'service_fee',
        'total',
        'paid_at',
        'midtrans_snap_token',
        'midtrans_transaction_id',
    ];

    protected $casts = [
        'subtotal' => 'integer',
        'service_fee' => 'integer',
        'total' => 'integer',
        'paid_at' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function branch(): BelongsTo
    {
        return $this->belongsTo(Branch::class);
    }

    public function items(): HasMany
    {
        return $this->hasMany(OrderItem::class);
    }

    /**
     * Get the order status label in Indonesian.
     */
    public function getOrderStatusLabelAttribute(): string
    {
        $labels = [
            'waiting_payment' => 'MENUNGGU BAYAR',
            'processing'      => 'DIPROSES',
            'making'          => 'DIBUAT',
            'ready'           => 'SIAP AMBIL',
            'completed'       => 'SELESAI',
            'cancelled'       => 'BATAL',
        ];

        return $labels[$this->order_status] ?? strtoupper($this->order_status);
    }

    /**
     * Get the payment status label in Indonesian.
     */
    public function getPaymentStatusLabelAttribute(): string
    {
        $labels = [
            'pending' => 'BELUM LUNAS',
            'paid'    => 'LUNAS',
            'failed'  => 'GAGAL',
            'expired' => 'KEDALUWARSA',
        ];

        return $labels[$this->payment_status] ?? strtoupper($this->payment_status);
    }
}
